daml-foundations/daml-ghc/daml-stdlib-src/DA/Internal/Template.daml

```diff
@@ -179,6 +179,10 @@ instance IsParties Party where
 instance IsParties [Party] where
   toParties ps = ps
 
+instance IsParties (Optional Party) where
```

Comment:
We could generalize this to `instance IsParties a => IsParties (Optional a)`. Not sure how sensible that would be.

Reply:
Actually I was thinking today that I will need `[Optional Party]` too. So guess they ought to both be generalized.

Reply:
@cocreature 
Hmm - something like this, perhaps?
```
instance IsParties Party where
  toParties p = [p]

instance IsParties a => IsParties [a] where
  toParties ps = concat $ fmap toParties ps

instance IsParties a => IsParties (Optional a) where
  toParties None = []
  toParties (Some p) = toParties p
```

Reply:
Looks sensible to me but it would probably be good if someone who is more familiar with this takes a look, maybe @bame-da or @hurryabit.

Reply:
Generalising the list version is going to lead to worse error messages. My inclination is that we should add `Optional Party` and `[Party]`, but if @sofus-da wants `[Optional Party]` then a catMaybes (or whatever we called it) should be used.